pkg/controller/migmigration/stage.go

```diff
@@ -91,6 +90,46 @@ func BuildStagePods(labels map[string]string,
 	return stagePods
 }
 
+// get the running application pods with attached PVC
+func GetApplicationPodsWithStageLabels(labels map[string]string, pvcMapping map[k8sclient.ObjectKey]migapi.PV, list *[]corev1.Pod) []corev1.Pod {
+	applicationPodsWithStageLabel := []corev1.Pod{}
+	for _, pod := range *list {
+		foundVolume := true
+		excludePVC := ""
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim == nil {
+				continue
+			}
+			claimKey := k8sclient.ObjectKey{
+				Name:      volume.PersistentVolumeClaim.ClaimName,
+				Namespace: pod.Namespace,
+			}
+			pv, found := pvcMapping[claimKey]
+			if !found {
+				continue
+			}
+			if pv.Selection.Action != migapi.PvCopyAction ||
+				pv.Selection.CopyMethod != migapi.PvFilesystemCopyMethod {
+				excludePVC += volume.Name + ";"
```

Comment:
Create excludePVC as a `[]string` instead of a `string` and build up a list instead of append here, that way the list doesn't end with a delimiter. Also, when you do build the string for the annotation, use `strings.Join(excludePVC, ",")` so it will be comma-delimited like the other restic PV annotations. Also, I think this is probably more appropriate as an annotation rather than a label.